### PR TITLE
Day 4 in Go

### PR DESCRIPTION
**Pull request by tholok97**

### CODE REVIEW

File: days/day-04/solutions/day04.tholok97.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	"strconv"
+	"strings"
+)
+
+// validator is any function that can tell you if the given string is valid
+// according to some criteria
+type validator func(string) bool
+
+// fieldValidators provides a validator for each key
+type fieldValidators map[string]validator
```
Comment (Arxcis) [inline days/day-04/solutions/day04.tholok97.go:17]: This is like a generic container. I thought golang did not have generics ?! :smile: You proved me wrong! Why is this not a generic container=? @hypirion help! My world-view is broken

**Review by Arxcis — APPROVED** [review]
Very clean solution :smile: Me like

File: days/day-04/solutions/day04.tholok97.go (hunk 1 above)
Comment (hypirion) [reply days/day-04/solutions/day04.tholok97.go:17]: Go _has_ parametric polymorphism (aka generics), but it's only for [builtin](https://godoc.org/builtin) types and functions.

So yeah, the `map` type here is a generic, but `fieldValidators` or any of the other types are most certainly not generic.

Comment (Arxcis) [reply days/day-04/solutions/day04.tholok97.go:17]: Aww. I see :+1: Thank you for the explanation

**Pull request merged**